llvm/include/llvm/ADT/ArrayRef.h

```diff
@@ -149,6 +149,20 @@ namespace llvm {
                  * = nullptr)
         : Data(Vec.data()), Length(Vec.size()) {}
 
+    /// Construct an ArrayRef<T> from iterator_range<U*>. This uses SFINAE
+    /// to ensure that this is only used for iterator ranges of random access
+    /// iterators that can be converted.
+    template <typename U,
+              typename = std::enable_if<
+                  std::is_base_of<
+                      std::random_access_iterator_tag,
+                      typename std::iterator_traits<
+                          decltype(std::declval<const iterator_range<U *> &>()
+                                       .begin())>::iterator_category>::value &&
```

Comment:
So what I meant is that we can factor this typename into a new template parameter, say IT, and then use IT here to keep this template parameter readable?